HeroVault/HeroVault/Features/Home/HomeScreenDetails/HomeScreenDetailsViewModel.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  HomeScreenDetailsViewModel.swift
+//  HeroVault
+//
+//  Created by Gia Catano on 2024/04/20.
+//
+
+import Foundation
+
+class HomeScreenDetailsViewModel {
+    
+    // MARK: Variables
+    
+    var marvelData: MarvelData?
```

Comment:
```suggestion
    private var marvelData: MarvelData?
```

Reply:
a lot of these can be private
    private var marvelData = [MarvelData]()
    private var filteredMarvelData = [MarvelData]()
    private var marvelDataType: EntityType
    private var error: Error?
    private var isSearching: Bool
    private let homeScreenRepository: HomeScreenRepositoryType?
    private weak var delegate: ViewModelProtocol?
    var hideNoResultsText: Bool